server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -0,0 +1,263 @@
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { InductionDto, PreviousWorkExperienceDto } from 'inductionDto'
+import type { PreviousWorkExperienceTypesForm } from 'inductionForms'
+import type { PrisonerSummary } from 'viewModels'
+import logger from '../../../../logger'
+import PreviousWorkExperienceTypesController from '../common/previousWorkExperienceTypesController'
+import { InductionService } from '../../../services'
+import validatePreviousWorkExperienceTypesForm from './previousWorkExperienceTypesFormValidator'
+import TypeOfWorkExperienceValue from '../../../enums/typeOfWorkExperienceValue'
+import toCreateOrUpdateInductionDto from '../../../data/mappers/createOrUpdateInductionDtoMapper'
+import previousWorkExperienceTypeScreenOrderComparator from '../previousWorkExperienceTypeScreenOrderComparator'
+
+export default class PreviousWorkExperienceTypesUpdateController extends PreviousWorkExperienceTypesController {
+  constructor(private readonly inductionService: InductionService) {
+    super()
+  }
+
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/work-and-interests`
+  }
+
+  getBackLinkAriaText(_req: Request): string {
+    return 'Back to <TODO - check what CIAG UI does here>'
+  }
+
+  submitPreviousWorkExperienceTypesForm: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary, inductionDto } = req.session
+
+    req.session.previousWorkExperienceTypesForm = { ...req.body }
+    if (!req.session.previousWorkExperienceTypesForm.typeOfWorkExperience) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = []
+    }
+    if (!Array.isArray(req.session.previousWorkExperienceTypesForm.typeOfWorkExperience)) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = [
+        req.session.previousWorkExperienceTypesForm.typeOfWorkExperience,
+      ]
+    }
+    const { previousWorkExperienceTypesForm } = req.session
+
+    const errors = validatePreviousWorkExperienceTypesForm(previousWorkExperienceTypesForm, prisonerSummary)
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      return res.redirect(`/prisoners/${prisonNumber}/induction/previous-work-experience`)
+    }
```

Comment:
Up to this point, the submit handler of this controller should be no different to any of the others we are doing in this epic. IE. get the form from the request body, fix up any array fields, and validate the form 👍 

The complexity comes next ! 🤣